app/src/main/java/org/jellyfin/androidtv/ui/browsing/BrowseGridFragment.java

```diff
@@ -863,13 +864,7 @@ public void onClick(View v) {
         mSettingsButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Intent settingsIntent = new Intent(getActivity(), PreferencesActivity.class);
-                settingsIntent.putExtra(PreferencesActivity.EXTRA_SCREEN, DisplayPreferencesScreen.class.getCanonicalName());
-                Bundle screenArgs = new Bundle();
-                screenArgs.putString(DisplayPreferencesScreen.ARG_PREFERENCES_ID, mFolder.getDisplayPreferencesId());
-                screenArgs.putBoolean(DisplayPreferencesScreen.ARG_ALLOW_VIEW_SELECTION, userViewsRepository.getValue().allowViewSelection(mFolder.getCollectionType()));
-                settingsIntent.putExtra(PreferencesActivity.EXTRA_SCREEN_ARGS, screenArgs);
-                requireActivity().startActivity(settingsIntent);
+                navigationRepository.getValue().navigate(Destinations.INSTANCE.displayPreferences(mFolder.getDisplayPreferencesId(), userViewsRepository.getValue().allowViewSelection(mFolder.getCollectionType())));
```

Comment:
Not sure what the actual line length is here, but it seems a bit long for me to parse easily.